src/runmode-dpdk.c

```diff
@@ -1021,7 +1021,8 @@ static void DeviceInitPortConf(const DPDKIfaceConfig *iconf,
 
     if (iconf->checksum_mode == CHECKSUM_VALIDATION_DISABLE) {
         SCLogConfig("%s: checksum validation disabled", iconf->iface);
-    } else if (dev_info->rx_offload_capa & RTE_ETH_RX_OFFLOAD_CHECKSUM) {
+    } else if ((dev_info->rx_offload_capa & RTE_ETH_RX_OFFLOAD_CHECKSUM) ==
```

Comment:
what is the fix here?

Reply:
It is a multibit comparison so this ensures all respective bits are properly set (and not just a subset)

```c
 #define RTE_ETH_RX_OFFLOAD_CHECKSUM (RTE_ETH_RX_OFFLOAD_IPV4_CKSUM | \
                  RTE_ETH_RX_OFFLOAD_UDP_CKSUM | \
                  RTE_ETH_RX_OFFLOAD_TCP_CKSUM)
```

Reply:
there should be a comment in the code then, to point it out as it is non-obvious

Reply:
ok, added a comment